File: src/test/java/com/raychenon/leetcode/RansomNoteCharTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-09-16
 * https://leetcode.com/problems/ransom-note/
 */
public class RansomNoteCharTest {

    private static RansomNoteChar instance;

    @BeforeClass
    public static void setUp() {
        instance = new RansomNoteChar();
    }

    @Test
    public void test1() {
        compareCanConstructRansomMethods(false, "a", "b");
    }

    @Test
    public void test2() {
        compareCanConstructRansomMethods(false, "aa", "ab");
    }

    @Test
    public void test3() {
        compareCanConstructRansomMethods(true, "aa", "aab");
    }

    public void compareCanConstructRansomMethods(boolean expected, String ransom, String magazine) {
        assertEquals(expected, instance.canConstruct(ransom, magazine));
        assertEquals(expected, instance.canConstructArray(ransom, magazine));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/RansomNoteCharTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-09-16
 * https://leetcode.com/problems/ransom-note/
 */
class RansomNoteCharTest {

    @Test
    fun testRansom1() {
        assertRansomConstructMethods(false, "a", "b")
    }

    @Test
    fun testRansom2() {
        assertRansomConstructMethods(false, "aa", "ab")
    }

    @Test
    fun testRansom3() {
        assertRansomConstructMethods(true, "aa", "aab")
    }

    fun assertRansomConstructMethods(expected: Boolean, ransom: String, magazine: String): Unit {
        Assert.assertEquals(expected, RansomNoteChar.canConstruct(ransom, magazine))
        Assert.assertEquals(expected, RansomNoteChar.canConstructArray(ransom, magazine))
    }
}